tests/ml/sklearn_test.py

```diff
@@ -160,3 +160,56 @@ def test_sklearn_estimator_classification_validation():
         skl_pred = model.predict(Xtest)
 
         assert np.all(skl_pred == test.pred.values)
+
+def test_sklearn_incremental_predictor():
+    df = vaex.example()
+    df_train, df_test = df.ml.train_test_split(test_size=0.1, verbose=False)
+
+    features = df_train.column_names[:6]
+    target = 'FeH'
+
+    incremental = IncrementalPredictor(model=SGDRegressor(),
+                                       features=features,
+                                       batch_size=10_000,
+                                       num_epochs=5,
+                                       shuffle=True,
+                                       prediction_name='pred')
+    incremental.fit(df=df_train, target=target)
+    df_train = incremental.transform(df_train)
+
+    # State transfer
+    state = df_train.state_get()
+    df_test.state_set(state)
+
+    assert df_train.column_count() == df_test.column_count()
+    assert df_test.pred.values.shape == (33000,)
+
+    pred_in_memory = incremental.predict(df_test)
+    np.testing.assert_array_almost_equal(pred_in_memory, df_test.pred.values, decimal=2)
+
+
+def test_sklearn_incremental_predictor_serialize(tmpdir):
+    df = vaex.example()
+    df_train, df_test = df.ml.train_test_split(test_size=0.1, verbose=False)
+
+    features = df_train.column_names[:6]
+    target = 'FeH'
+
+    incremental = IncrementalPredictor(model=SGDRegressor(),
+                                       features=features,
+                                       batch_size=10_000,
```

Comment:
Make batchsize an odd number and see if you can check from the skmodel how many samples it has seen.

Reply:
Unfortunately the sklearn models do not keep track on the number of samples they have seen.. however I can test this manually..

Reply:
You could use mock to wrap the method, e.g.:
https://stackoverflow.com/questions/18869141/using-python-mock-to-spy-on-calls-to-an-existing-object

Reply:
Added an additional test uses `mock` to count the number of calls to the underlying `partial_fit` method to the model.